jina/clients/base/stream_rpc.py

```diff
@@ -0,0 +1,68 @@
+from jina.clients.helper import callback_exec
+from jina.proto import jina_pb2_grpc
+
+
+class StreamRpc:
+    """Class that encapsulated the methods required to run a stream rpc call from the client. Instantiate a single class
+    for each client request.
+    """
+
+    def __init__(
+        self,
+        channel,
+        continue_on_error,
+        metadata,
+        on_always,
+        on_done,
+        on_error,
+        p_bar,
+        req_iter,
+        max_attempts,
+        backoff_multiplier,
+        initial_backoff,
+        max_backoff,
+        logger,
+        show_progress,
+        compression,
+        **kwargs
+    ):
+        self.compression = compression
+        self.show_progress = show_progress
+        self.logger = logger
+        self.max_backoff = max_backoff
+        self.initial_backoff = initial_backoff
+        self.backoff_multiplier = backoff_multiplier
+        self.max_attempts = max_attempts
+        self.req_iter = req_iter
+        self.p_bar = p_bar
+        self.on_error = on_error
+        self.on_done = on_done
+        self.on_always = on_always
+        self.metadata = metadata
+        self.continue_on_error = continue_on_error
+        self.channel = channel
+        self.kwargs = kwargs
+
+    async def stream_rpc_with_retry(self):
```

Comment:
`stream_rpc_with_retry` is misleading. As you described, the stream RPC does not support retry.

Reply:
It's still valid because the grpc.serviceoptions are still enabled and the retry might work if certain conditions are met. If the documentation is not clear, I will update it.